File: pet-clinic-data/src/main/java/com/sachet/sfgpetclinic/services/SpecialityService.java
package com.sachet.sfgpetclinic.services;

import com.sachet.sfgpetclinic.model.Speciality;

import java.util.Set;

public interface SpecialityService {

    Set<Speciality> findAll();

    Speciality findById(Long id);

    Speciality save(Speciality speciality);

    void delete(Speciality speciality);

    void deleteById(Long id);

}

File: src/main/kotlin/com/sachet/sfgpetclinic/service/SpecialityService.kt
package com.sachet.sfgpetclinic.service

import com.sachet.sfgpetclinic.model.PetType
import com.sachet.sfgpetclinic.model.Speciality

interface SpecialityService {

    fun findById(id: Long): Speciality
    fun findAll(): Set<Speciality>
    fun save(petType: PetType): PetType
    fun delete(petType: PetType)
    fun deleteById(id: Long)

}